### DISCUSSION TITLE
[Question][Hw2] Order of a model of satisfiable case

### DISCUSSION THREAD

**Discussion by pingpingy1** [prosyslab-classroom/cs424-program-reasoning#370] [Q&A]
Hello.

I have been implementing the solvers for Hw2.
I believe I have correctly implemented the program equivalence checker, in that I think it prints "satisfiable" along with a model when the programs are not equivalent, and "unsatisfiable" when they are equivalent.

However, for some reason, when printing a model in which the programs give different results, my implementation prints the variables in a different order than was provided on Github.

Below is a picture of my testing with the eq3.scm file.
Instead of printing in `tgt_x` -> `src_x` -> `src_y` order,
it does in `tgt_x` -> `src_y` -> `src_x` order.
Moreover, when testing with `make test`, this is registered as a wrong output.

My question is, does the order of printing a satisfying model matter?
Should I modify my code so that it correctly prints in the designated order?

Thank you.

![image](https://user-images.githubusercontent.com/94441510/192955080-82226b48-1217-455b-ad8e-75576a45765b.png)

**Comment by Lee-Janggun** [comment]
See #47

**Comment by pingpingy1** [comment]
Oh, sorry for the duplicate question.
Thanks.

**Comment by KihongHeo** [comment]
Thanks. Janggun.
It would be nice for us to have an AI-based TA bot that automatically points out existing answers for duplicated questions. 😄
